add_pricelist_price/models/sale_order.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    # computational field for book price
+    book_price = fields.Float(
+        string="Book Price", 
+        compute="_compute_book_price", 
+        help="Price without any discount and taxes",
+    )
+
+    @api.depends('product_id', 'product_uom_qty', 'order_id.pricelist_id')
+    def _compute_book_price(self):
+        # function for computing book price depends on product, quantity and pricelist
+        for line in self:
+            if line.product_id and line.order_id:
+                book_price = line.order_id.pricelist_id._get_product_price(
+                    product=line.product_id, 
+                    quantity=line.product_uom_qty,
+                    uom=line.product_uom, # unit of measure
+                )
+                line.book_price = book_price or line.product_id.lst_price
```

Comment:
```suggestion
                line.book_price = book_price
```
`_get_product_price` will always return float please see the doc string of the method

Reply:
Got it just saw it.